ledger/acctupdates.go

```diff
@@ -1971,6 +1974,61 @@ func (au *accountUpdates) commitRound(offset uint64, dbRound basics.Round, lookb
 
 }
 
+// compactDeltas takes an arary of account map deltas ( one array entry per round ), and corresponding creatables array, and compact the arrays into a single
+// map that contains all the account deltas changes. While doing that, the function eliminate any intermediate account changes.
+// The function returns a map that could be the same as the input map, and therefore both the outputs of this function as well as the inputs to this function
+// should be kept read-only ( unless we don't care about the correctness of the data anymore ).
+func compactDeltas(accountDeltas []map[basics.Address]accountDelta, creatableDeltas []map[basics.CreatableIndex]modifiedCreatable) (outAccountDeltas map[basics.Address]accountDelta, outCreatableDeltas map[basics.CreatableIndex]modifiedCreatable) {
+	// this is a common case where we have only a single round that we try to flush. In this case, we're just going to return
+	// the first entry.
+	if len(accountDeltas) == 1 && len(creatableDeltas) == 1 {
+		outAccountDeltas = accountDeltas[0]
+		outCreatableDeltas = creatableDeltas[0]
+		return
+	}
+
+	if len(accountDeltas) > 0 {
+		// the sizes of the maps here aren't super accurate, but would hopefully be a rough estimate for a resonable starting point.
+		outAccountDeltas = make(map[basics.Address]accountDelta, 1+len(accountDeltas[0])*len(accountDeltas))
+		for _, roundDelta := range accountDeltas {
+			for addr, acctDelta := range roundDelta {
+				if prev, has := outAccountDeltas[addr]; has {
+					outAccountDeltas[addr] = accountDelta{
+						old: prev.old,
+						new: acctDelta.new,
+					}
+				} else {
+					// it's a new entry.
+					outAccountDeltas[addr] = acctDelta
+				}
+			}
+		}
+	}
+
+	if len(creatableDeltas) > 0 {
+		// the sizes of the maps here aren't super accurate, but would hopefully be a rough estimate for a resonable starting point.
+		outCreatableDeltas = make(map[basics.CreatableIndex]modifiedCreatable, 1+len(creatableDeltas[0])*len(creatableDeltas))
+		for _, roundCreatable := range creatableDeltas {
+			for creatableIdx, creatable := range roundCreatable {
+				if prev, has := outCreatableDeltas[creatableIdx]; has {
+					outCreatableDeltas[creatableIdx] = modifiedCreatable{
+						ctype:   creatable.ctype,
+						created: creatable.created,
+						creator: creatable.creator,
+						// The following ndeltas is not currently used
+						// TODO : it should be "converted" to counting the number of updates for a particular creatable entry and replace the creatableFlushcount in commitRound.
+						ndeltas: creatable.ndeltas + prev.ndeltas,
```

Comment:
they are set in `newBlockImpl` and checked in `commitRound`. Shouldn't `ndeltas` increment per seen entry instead of summing?

Reply:
as with the comment, the content of this field is not being used *at all*. I will provide proper implementation on a subsequent PR, where these would be used instead the existing flushCount. ( but we're not there yet - that's why I added the comment ).

Reply:
and yes, it will become a counter of how many updates we had across the rounds.